crates/gitbutler-repo/src/hooks.rs

```diff
@@ -0,0 +1,86 @@
+use std::path::PathBuf;
+
+use anyhow::Result;
+use git2_hooks;
+use git2_hooks::HookResult as H;
+use gitbutler_command_context::CommandContext;
+use gitbutler_diff::GitHunk;
+use serde::Serialize;
+
+use crate::staging;
+
+#[derive(Serialize, PartialEq, Debug, Clone)]
+pub struct MessageData {
+    pub message: String,
+}
+
+#[derive(Serialize, PartialEq, Debug, Clone)]
+pub struct ErrorData {
+    pub error: String,
+}
+
+/// Hook result indicating either success or failure.
+#[derive(Serialize, PartialEq, Debug, Clone)]
+#[serde(tag = "status", rename_all = "lowercase")]
+pub enum HookResult {
```

Comment:
What is the difference between Message and Success here? Or Failure and NotFound? 
Can we simplify this enum?

Reply:
- `NotFound` means no hook executed
- `Failure` means hook ran, but failed
- `Message` is special for `commit-msg`, if it modifies message
- `Success` means what you think it means

I wouldn't really want to simplify, since this setup makes it easier to conditionally display small toasts saying e.g. "Pre-commit hook successful", or "Hook modified your commit message". What do you think?

Reply:
I've renamed it to `NotConfigured`.